src/service.ts

```diff
@@ -1,50 +1,199 @@
 import { INestApplicationContext, Injectable } from '@nestjs/common';
-import * as ora from 'ora';
-import { ICommand, forwardSubCommands } from './commander';
-import { InjectCommander } from './decorators';
-import { IWithApplicationContext } from './interfaces';
+import { Command, CommanderError } from 'commander';
+import { CommandError, onSubCommand, createCli } from './commander';
+import {
+    IConsoleOptions,
+    InjectCommander,
+    ICommandDecoratorOptions
+} from './decorators';
+import { formatResponse } from './helpers';
+import { EventEmitter } from 'events';
 
 @Injectable()
-export class ConsoleService implements IWithApplicationContext {
+export class ConsoleService {
     protected container: INestApplicationContext;
+    protected commands: Map<string, Command> = new Map();
+    protected eventManager: EventEmitter = new EventEmitter();
 
-    constructor(@InjectCommander() protected readonly cli: ICommand) {}
+    constructor(@InjectCommander() protected cli: Command) {}
 
-    static createSpinner(text?: string) {
-        return ora.default(text);
+    /**
+     * Reset the cli stack (for testing purpose only)
+     */
+    resetCli() {
+        this.cli = createCli();
     }
 
-    getCli() {
-        return this.cli;
+    /**
+     * Log an error
+     * @param command The command related to the error
+     * @param error The error to format
+     */
+    logError(command: Command, error: Error) {
+        // tslint:disable-next-line:no-console
+        return console.error(formatResponse(error));
     }
 
-    setContainer(container: INestApplicationContext): IWithApplicationContext {
+    /**
+     * Get a cli
+     * @param name Get a cli by name, if not set, the root cli is used
+     */
+    getCli(name?: string) {
+        return name ? this.commands.get(name) : this.cli;
+    }
+
+    /**
+     * Set the container
+     */
+    setContainer(container: INestApplicationContext): ConsoleService {
         this.container = container;
         return this;
     }
 
+    /**
+     * Get the container
```

Comment:
Why call this the container and not something like `appContext`? In the nestjs code, the container is the the Nestjs Container https://github.com/nestjs/nest/blob/328c33c957a47f63ef07de32bef06bc57a04d8ad/packages/core/injector/container.ts#L20 which doesn't seem to be really exposed.

Reply:
Its true. But as we use the application like a container (eg app.get) and the NestContainer is a private api, it seems legit to call it container.That said I am not against the appContext, and it would be clear for user that this is an app. Note the with the v2, it could be an app instance (NestApplication) and not only an app context (INestApplicationContext)